examples/rdl.py

```diff
@@ -0,0 +1,710 @@
+
+    # Print table information for debugging
+    print(f"Train table columns: {train_table.df.columns}")
+    print(f"Val table columns: {val_table.df.columns}")
+    print(f"Test table columns: {test_table.df.columns}")
+
+    for split, table in [
+        ("train", train_table),
+        ("val", val_table),
+        ("test", test_table),
+    ]:
+        table_input = get_node_train_table_input(
+            table=table,
+            task=task,
+        )
+        entity_table = table_input.nodes[0]
+
+        # Create a dictionary mapping edge types to number of neighbors
+        num_neighbors_dict = {
+            edge_type: args.num_neighbors  # [num_neighbors for each direction]
+            for edge_type in data.edge_types
+        }
+
+        loader_dict[split] = NeighborLoader(
+            data=data,
+            num_neighbors=num_neighbors_dict,
+            input_nodes=table_input.nodes,
+            input_time=table_input.time,
+            time_attr="time",
+            transform=table_input.transform,
+            batch_size=args.batch_size,
+            temporal_strategy=args.temporal_strategy,
+            shuffle=split == "train",
+            num_workers=args.num_workers,
+            persistent_workers=args.persistent_workers,
+        )
+
+    # Get task-specific parameters
+    print("Getting task-specific parameters...")
+    (out_channels, loss_fn, tune_metric,
+     higher_is_better) = get_task_type_params(task)
+    print("out_channels: ", out_channels)
+    print("loss_fn: ", loss_fn)
+    print("tune_metric: ", tune_metric)
+    print("higher_is_better: ", higher_is_better)
+
+    # Define the model
+    print("Initializing the model...")
+    model = Model(
+        data=data,
+        col_stats_dict=col_stats_dict,
+        num_layers=len(args.num_neighbors),
+        channels=args.channels,
+        out_channels=args.out_channels,
+        aggr=args.aggr,
+        norm=args.norm,
+    ).to(device)
+
+    # Define the optimizer
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.learning_rate)
+    """ Train the model """
+    print("Training the model...")
+    state_dict = None
+    best_val_metric = -math.inf if higher_is_better else math.inf
+    for epoch in range(1, args.epochs + 1):
+        train_loss = train(
+            model=model,
+            loader_dict=loader_dict,
+            task=task,
+            optimizer=optimizer,
+            loss_fn=loss_fn,
+        )
+        val_pred = test(
+            loader=loader_dict["val"],
+            model=model,
+            task=task,
+        )
+        val_metrics = task.evaluate(val_pred, val_table)
+        print(f"Epoch: {epoch:02d}, "
+              f"Train loss: {train_loss}, "
+              f"Val metrics: {val_metrics}")
+
+        if (higher_is_better and val_metrics[tune_metric] > best_val_metric
+            ) or (not higher_is_better
+                  and val_metrics[tune_metric] < best_val_metric):
```

Comment:
```suggestion
        if ((higher_is_better and val_metrics[tune_metric] > best_val_metric
            ) or (not higher_is_better
                  and val_metrics[tune_metric] < best_val_metric)):
```
for nicer formatting.